### PR TITLE
subscribe button & form not render if is disabled #818

### LINKED ISSUE #818: MT Market Sync: Subscribe form not integrated
### Issue Type

Defect

### Summary

On all market sites the subscribe form is not integrated.

### Priority

High

### Description

On MT market sites the subscribe form is not integrated and appears only as text on the page.

### Affected Device Classes

Mobile, Tablet, Desktop

### Affected Browsers/Devices and Versions

N/A

### Steps to Reproduce

1. Navigate to a MT market site
2. Navigate to Bulldog Magazine 
3. Scroll down to subscribe 
4. Observe a missing integration to be able to subscribe

### Expected Results

A form is available so that as a user I can subscribe to Bulldog Magazine articles

Image

### Actual Results

Image

### Attachment

_No response_

### PR DESCRIPTION
**Pull request by jonatan-lledo-netcentric**
Fix #818

blocks affected:
- sub-nav (check if the object MAGAZINE_CONFIG has the attribute HREF and also if it has an element with the dataset form-type and if both are missing then doesn't render the buttons)
- magazine-subscribe (code improvement)
- iframe (updated with createElement fn)

Test URLs:
### magazine
- Before: https://main--vg-macktrucks-com--hlxsites.hlx.page/magazine/
- After: https://818-mt-mkt-sync-subscribe-form--vg-macktrucks-com--hlxsites.hlx.page/magazine/
### explore-articles
- Before: https://main--vg-macktrucks-com--hlxsites.hlx.page/magazine/explore-articles
- After: https://818-mt-mkt-sync-subscribe-form--vg-macktrucks-com--hlxsites.hlx.page/magazine/explore-articles
### example magazine article
- Before: https://main--vg-macktrucks-com--hlxsites.hlx.page/magazine/articles/2022/august/expertise-in-action-preview
- After: https://818-mt-mkt-sync-subscribe-form--vg-macktrucks-com--hlxsites.hlx.page/magazine/articles/2022/august/expertise-in-action-preview
### example magazine article without the magazine-subscribe block
- Before: https://main--vg-macktrucks-com--hlxsites.hlx.page/drafts/jlledo/magazine/expertise-in-action-preview
- After: https://818-mt-mkt-sync-subscribe-form--vg-macktrucks-com--hlxsites.hlx.page/drafts/jlledo/magazine/expertise-in-action-preview

### CODE REVIEW

File: blocks/sub-nav/sub-nav.js [hunk 1]
```diff
@@ -130,32 +132,45 @@ async function buildMagazineSubNav(block, ref) {
...
   innerList.className = 'sub-nav-list inner';
-  listSubscribeBtn.textContent = subscribeText;
-  supR.textContent = '®';
-  listSubscribeBtn.appendChild(supR);
-  listSubscribeBtnContainer.appendChild(listSubscribeBtn);
   listWrapper.appendChild(closeBtn);
-  listContainer.append(listWrapper, listSubscribeBtnContainer);
+
+  if (isSubscribeEnabled) {
+    listSubscribeBtn.textContent = subscribeText;
```
Comment (alexiscoelho) [inline blocks/sub-nav/sub-nav.js:138]: listSubscribeBtnContainer and listSubscribeBtn can move inside this if block, right?

File: blocks/sub-nav/sub-nav.js [hunk 2]
```diff
@@ -130,32 +132,45 @@ async function buildMagazineSubNav(block, ref) {
...
   mainTitleLink.title = mainTitleImg.lastElementChild.alt;
-  subscribeBtn.textContent = subscribeText;
-  subscribeBtnContainer.appendChild(subscribeBtn);
   mainTitleLink.appendChild(mainTitleImg);
   subNavTitle.appendChild(mainTitleLink);
   mainSubNav.appendChild(subNavTitle);
-  subNavContainer.append(listIcon, mainSubNav, subscribeBtnContainer, listContainer);
+
+  if (isSubscribeEnabled) {
+    subscribeBtn.textContent = subscribeText;
```
Comment (alexiscoelho) [inline blocks/sub-nav/sub-nav.js:158]: Also the `const subscribeBtn` here?

**Review by alexiscoelho — CHANGES_REQUESTED** [review]

File: blocks/sub-nav/sub-nav.js (hunk 2 above)
Comment (jonatan-lledo-netcentric) [reply blocks/sub-nav/sub-nav.js:158]: done

File: blocks/sub-nav/sub-nav.js (hunk 1 above)
Comment (jonatan-lledo-netcentric) [reply blocks/sub-nav/sub-nav.js:138]: done

**Review by alexiscoelho — APPROVED** [review]

**Review by santi-homps — APPROVED** [review]

**Pull request merged**